1-Draft/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,446 @@
+﻿---
+RFC: RFCxxxx
+Author: Sydney Smith
+Status: Draft
+SupercededBy: N/A
+Version: 0.3
+Area: PowerShellGet
+Comments Due: 6/30
+Plan to implement: Yes, PS7
+---
+
+# PowerShellGet 3.0 Module
+
+PowerShellGet has enabled an active community to publish and install PowerShell
+Modules.
+However, it has some architectural and design limitations that prevent it from
+moving forward more quickly.
+This RFC proposes some significant changes to address this.
+
+> [!NOTE]
+> There are no intentions to break existing PSGet v2 users so existing
+> users can stay on PSGet v2 and that will continue to work.
+## Motivation
+
+    As a PowerShell user,
+    I can discover how to install missing cmdlets,
+    so that I can be more productive.
+
+    As a PowerShell user,
+    I can easily install modules without specifying lots of switches,
+    so that I can be more productive.
+
+    As a PowerShellGet contributor,
+    I can more easily contribute code,
+    so that PowerShellGet evolves more quickly.
+
+## User Experience
+
+```powershell
+PS> Get-Satisfaction
+Get-Satisfaction : The term 'Get-Satisfaction' is not recognized as the name of a cmdlet, function, script file, or operable program.
+Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
+At line:1 char:1
++ Get-Satisfaction
++ ~~~
++ CategoryInfo          : ObjectNotFound: (Get-Satisfaction:String) [], CommandNotFoundException
++ FullyQualifiedErrorId : CommandNotFoundException
+Suggestion: You can obtain `Get-Satisfaction` by running `Install-PSResource HappyDays`.
+PS> Install-PSResource HappyDays
+```
+
+## Specification
+
+### Rewrite of PowerShellGet
+
+PowerShellGet is currently written as PowerShell script with a dependency on PackageManagement.
+Proposal is to write PSResource in C# to reduce complexity and make easier to maintain.
+In addition, remove dependency on PackageManagement completely as well as dependency on
+nuget.exe.
+This module will depend on https://www.nuget.org/packages/NuGet.Client.
+This module would be shipped in the PSGallery supporting older
+versions of PowerShell.
+This module will ship in PowerShell 7.1.

```

Comment:
Please clarify: Will it ship in addition to the exiting PowerShellGet 2.x or be a replacement? I understand they can work side-by-side but knowing what the 'out of the box' modules are is important for CI scenarios where installation of certain modules is assumed. Since the cmdlet names are different, removing PowerShellGet 2.x would be a big breaking change

Reply:
The plan is to ship in addition to PowerShellGet 2.x for the 7.1 timeframe, and before that we will be shipping it in the Gallery as a standalone module to work out issues before we ship in PowerShell